applications/DEMApplication/python_scripts/sphere_strategy.py

```diff
@@ -319,12 +320,41 @@ def SolveSolutionStep(self):
         self.FixExternalForcesManually(time)
         (self.cplusplus_strategy).Solve()
 
-    def AdvanceInTime(self, step, time):
+    def AdvanceInTime(self, step, time, is_time_to_print = False):
         step += 1
-        time = time + self.dt
-
+        time += self.dt
+        self.UpdateTimeInModelParts(time, step, is_time_to_print)
         return step, time
 
+    def MoveAllMeshes(self, time, dt):
+
+        spheres_model_part = self.all_model_parts.Get("SpheresPart")
+        DEM_inlet_model_part = self.all_model_parts.Get("DEMInletPart")
+        rigid_face_model_part = self.all_model_parts.Get("RigidFacePart")
+        cluster_model_part = self.all_model_parts.Get("ClusterPart")
+
+        self.mesh_motion.MoveAllMeshes(rigid_face_model_part, time, dt)
+        self.mesh_motion.MoveAllMeshes(spheres_model_part, time, dt)
+        self.mesh_motion.MoveAllMeshes(DEM_inlet_model_part, time, dt)
+        self.mesh_motion.MoveAllMeshes(cluster_model_part, time, dt)
+
+    def UpdateTimeInModelParts(self, time, step, is_time_to_print = False):
+        spheres_model_part = self.all_model_parts.Get("SpheresPart")
+        cluster_model_part = self.all_model_parts.Get("ClusterPart")
+        DEM_inlet_model_part = self.all_model_parts.Get("DEMInletPart")
+        rigid_face_model_part = self.all_model_parts.Get("RigidFacePart")
+
+        self.UpdateTimeInOneModelPart(spheres_model_part, time, step, is_time_to_print)
+        self.UpdateTimeInOneModelPart(cluster_model_part, time, step, is_time_to_print)
+        self.UpdateTimeInOneModelPart(DEM_inlet_model_part, time, step, is_time_to_print)
+        self.UpdateTimeInOneModelPart(rigid_face_model_part, time, step, is_time_to_print)
+
+    def UpdateTimeInOneModelPart(self, model_part, time, step, is_time_to_print = False):
```

Comment:
initial underscore

Reply:
These methods are not newly created. They are quite old, from a time when the Analysis stage did not exist, and we had not adopted these rules yet. Changing the name of some of these methods should be done carefully, maybe with warnings for some time, and I would dedicate a specific PR for that (we don't know who is calling them from private applications).